crawler.rb

```diff
@@ -33,16 +33,14 @@ def start_crawling()
       limit: @max_crawls, ignore_links: @crawlable.ignore_links,
       links: @crawlable.links) do |spider|
 
-      spider.every_html_page do |page|
-        url = page.url.to_s
-        raw_page = Nokogiri::HTML(page.doc.to_s)
+      spider.every_html_page do |raw_page|
 
-        crawled_page = CrawlablePages::CrawledPage.new(url=url)
-        crawled_page.title = page.title
+        crawled_page = CrawlablePages::CrawledPage.new(url= raw_page.url.to_s)
+        crawled_page.title = raw_page.title
 
         # searchs for the main components needed in crawlable object
         @crawlable.main_divs.each do |search_for|
-          parsed_page = raw_page.xpath(search_for).text.to_s
+          parsed_page = raw_page.search(search_for).text.to_s
```

Comment:
I think search() returns an Array. Can we do sth like raw_page.search(@crawlable.main_divs) without the loop, and then just .join() the array with a space delimiter?